File: Data Structures & Algorithms/sqrtx/submission-2.py
class Solution:
    def mySqrt(self, x: int) -> int:
        l = 0 
        r = x 
        while l <= r :
            m = (l+r)//2  #4
            value = m * m 
            if value > x :
                r = m -1 
            elif value < x :
                l = m+1
            else:
                return m 
        return r 

#  0  1  2  3  4  5  6  7  8       res = 2
#  

File: Data Structures & Algorithms/permutations/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[][]}
     */
    permute(nums) {
        let res = [] 

        function backtrack(path,index) {
            if (path.length == nums.length) {
                res.push([...path]) 
                return 
            } 

            for (let i = 0 ; i < nums.length ; i++) {
                if (path.includes(nums[i])){
                    continue
                }
                path.push(nums[i])
                backtrack(path,i+1) 
                path.pop()
             }
        }
        backtrack([],0) 
        return res
    }
}
